app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,110 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
```

Comment:
So it depends on whether your topic is concrete or abstract. If it is abstract you should be authorising.